### PR TITLE
Fix s.sm.duplex and security refs

### PR DESCRIPTION
**Pull request by StephenBonikowsky**
This is regarding #1839
Until PR #1894 is merged and I re-base this PR will also show all of those changes.

### CODE REVIEW

**Comment by StephenBonikowsky** [comment]
@ericstj with my changes based on looking at your old CoreFx PR #[corefx\6902](https://github.com/dotnet/corefx/pull/6902) I'm getting...

`C:\OSS\WCF\Tools\Packaging.targets(1097,5): error : PackageIndex from C:\OSS\WCF\pkg\baseline\\packageIndex.json, C:\OSS\WCF\packages\Microsoft.Private.PackageBaseline\2.0.0-preview2-25304-02\build\packageIndex.json is has an assembly version entry(s) for 4.0.3.0 which are no longer in package System.ServiceModel.Duplex.  Please run /t:UpdatePackageIndex on this project to commit an update. [C:\OSS\WCF\src\System.ServiceModel.Duplex\pkg\System.ServiceModel.Duplex.pkgproj]`

@zhenlan and I discussed this but we don't understand enough of the 'why' this is happening to figure out what the solution is.

File: src/System.ServiceModel.Security/ref/System.ServiceModel.Security.csproj [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
+    <AssemblyVersion>4.0.2.0</AssemblyVersion>
```
Comment (ericstj) [inline src/System.ServiceModel.Security/ref/System.ServiceModel.Security.csproj]: You could just change https://github.com/StephenBonikowsky/wcf/blob/45229d8763582d6e8ff8edb1bd1fc12db33af575/src/System.ServiceModel.Security/dir.props#L5

**Comment by zhenlan** [comment]
> you can either continue to move forward on versioning and continue to omit the desktop ref since it will unify at runtime so long as it stays at 4.0.x.y

Thanks @ericstj! At this point, I am leaning towards this option. We have multiple desktop ref versions out there already. Customers who ran into this can work around by removing binding redirect. So far we did not hear any complaint about this and we can revisit it if this becomes an issue for people in the future.

**Comment by StephenBonikowsky** [comment]
Thanks for your comments on this Eric, understanding this a little better now. We realized that we've actually changed the Duplex contract which we believe requires a minor version update.
Since what I was initially trying to do with this PR is pretty tightly connected to PR #1894 and what is currently here is completely wrong :) I am going to abandon this PR and add the changes we thing are necessary to PR #1894.

**Pull request closed without merging**